src/views/Pools/components/PoolsTable/ActionPanel/Stake.tsx

```diff
@@ -0,0 +1,268 @@
+  const [requestedApproval, setRequestedApproval] = useState(false)
+  const { onApprove } = useSousApprove(stakingTokenContract, sousId)
+  const { toastSuccess, toastError } = useToast()
+
+  const isBnbPool = poolCategory === PoolCategory.BINANCE
+  const allowance = userData?.allowance ? new BigNumber(userData.allowance) : BIG_ZERO
+  const stakedBalance = userData?.stakedBalance ? new BigNumber(userData.stakedBalance) : BIG_ZERO
+  const isNotVaultAndHasStake = !isAutoVault && stakedBalance.gt(0)
+
+  const stakingTokenBalance = userData?.stakingTokenBalance ? new BigNumber(userData.stakingTokenBalance) : BIG_ZERO
+  const stakingTokenPrice = useBusdPriceFromToken(stakingToken.symbol)
+  const stakingTokenPriceAsNumber = stakingTokenPrice && stakingTokenPrice.toNumber()
+
+  const stakedTokenBalance = getBalanceNumber(stakedBalance, stakingToken.decimals)
+  const stakedTokenDollarBalance = getBalanceNumber(
+    stakedBalance.multipliedBy(stakingTokenPrice),
+    stakingToken.decimals,
+  )
+
+  const {
+    userData: { userShares },
+    pricePerFullShare,
+  } = useCakeVault()
+
+  const { cakeAsBigNumber, cakeAsNumberBalance } = convertSharesToCake(userShares, pricePerFullShare)
+  const hasSharesStaked = userShares && userShares.gt(0)
+  const isVaultWithShares = isAutoVault && hasSharesStaked
+  const stakedAutoDollarValue = getBalanceNumber(cakeAsBigNumber.multipliedBy(stakingTokenPrice), stakingToken.decimals)
+
+  const needsApproval = !allowance.gt(0) && !isBnbPool
+
+  const [onPresentTokenRequired] = useModal(<NotEnoughTokensModal tokenSymbol={stakingToken.symbol} />)
+
+  const [onPresentStake] = useModal(
+    <StakeModal
+      isBnbPool={isBnbPool}
+      pool={pool}
+      stakingTokenBalance={stakingTokenBalance}
+      stakingTokenPrice={stakingTokenPriceAsNumber}
+    />,
+  )
+
+  const [onPresentVaultStake] = useModal(<VaultStakeModal stakingMax={stakingTokenBalance} pool={pool} />)
+
+  const [onPresentUnstake] = useModal(
+    <StakeModal
+      stakingTokenBalance={stakingTokenBalance}
+      isBnbPool={isBnbPool}
+      pool={pool}
+      stakingTokenPrice={stakingTokenPriceAsNumber}
+      isRemovingStake
+    />,
+  )
+
+  const [onPresentVaultUnstake] = useModal(<VaultStakeModal stakingMax={cakeAsBigNumber} pool={pool} isRemovingStake />)
+
+  const onStake = () => {
+    if (isAutoVault) {
+      onPresentVaultStake()
+    } else {
+      onPresentStake()
+    }
+  }
+
+  const onUnstake = () => {
+    if (isAutoVault) {
+      onPresentVaultUnstake()
+    } else {
+      onPresentUnstake()
+    }
+  }
+
+  const { targetRef, tooltip, tooltipVisible } = useTooltip(
+    t("You've already staked the maximum amount you can stake in this pool!"),
+    { placement: 'bottom' },
+  )
+
+  const reachStakingLimit = stakingLimit.gt(0) && userData.stakedBalance.gte(stakingLimit)
+
+  const handleApprove = useCallback(async () => {
```

Comment:
This should maybe be abstracted out of this component & shared with the card ui components rather than duplicating?